### PR TITLE
[DOC-4748] Add ARM64 download links for Linux

### PR DESCRIPTION
**Pull request by mdlinville**
## HTML Previews
[Releases index](https://deploy-preview-15035--cockroachdb-docs.netlify.app/docs/releases/index.html) (the diff is difficult to review)
[Releases 22.2 minor version page with ARM downloads](https://deploy-preview-15035--cockroachdb-docs.netlify.app/docs/releases/v22.2.html)
[Releases 22.1 minor version page without ARM downloads](https://deploy-preview-15035--cockroachdb-docs.netlify.app/docs/releases/v22.1.html)
[Installation](https://deploy-preview-15035--cockroachdb-docs.netlify.app/docs/v22.2/install-cockroachdb-linux.html)

## Changes
- Update Linux installation instructions to provide info about ARM
- Add linux_arm field to _data/releases.csv
- Add a Javascript confirm on click of each Experimental download, both on the Releases index and on minor-version pages (all Windows, and Linux ARM)
- Add logic to releases/index.md and _includes/releases/release-downloads-docker-image.md (which actually builds out all of the minor-release pages, despite its name) to display the new links
- Define Testing and Experimental at the top level of the Releases Index page
- Completely refactor releases/index.md in other ways:
    - Change the looping logic to generate separate tables per version per platform
    - Always show SHA256 links
    - Improve link text for downloadable versions, Github links, and source code
    - For Docker, add a link to the cockroachdb image tags page
    - Remove CSS classes that were adding extra padding
 
## Tests
- [x] Local build
- [x] Link check
- [x] Download links download the correct things

### CODE REVIEW

File: releases/index.md
Comment (mdlinville) [inline releases/index.md:91]: Needs improvement

File: releases/index.md
Comment (mdlinville) [inline releases/index.md:79]: This should probably be an internal table, but I ran into CSS issues with that approach.

**Review by nickvigilante — COMMENTED** [review]
The one row per architecture version seems like wasted space. Since you went through the effort of splitting each OS into its own table, I'm gonna try sending a commit on this PR that uses `rowspan` and `colspan` to split the Download column into two. We can then just probably use that logic for Mac when the Silicon builds come out.

**Comment by rmloveland** [comment]
@mdlinville I suggest looping in someone from the Marketing team as well (whoever is in charge of tracking our download stats).  

When I made some updates to this page for the v20.2 release, I inadvertently broke some stuff that tracks downloads on this page and some folks rely on it.  If the tracking still exists (no idea, but probably?) it needs to be confirmed as still working.

(PS I quite like the changes FWIW, just wanted to drop that info)

**Review by nickvigilante — APPROVED** [review]
Just make sure to squash everything prior to merging, but LGTM!

**Comment by mdlinville** [comment]
> @mdlinville I suggest looping in someone from the Marketing team as well (whoever is in charge of tracking our download stats).
> 
> When I made some updates to this page for the v20.2 release, I inadvertently broke some stuff that tracks downloads on this page and some folks rely on it. If the tracking still exists (no idea, but probably?) it needs to be confirmed as still working.
> 
> (PS I quite like the changes FWIW, just wanted to drop that info)

This seems to be lost to history. I guess if it is still an issue we might find out after we merge, but nobody can remember / find exactly what happened.

File: releases/index.md
Comment (mdlinville) [inline releases/index.md:19]: @mwang1026 PTAL at this language and the line below, per @kenliu-crl

Comment (mwang1026) [reply releases/index.md:19]: Overall makes sense. Is it right to assume that we're labeling ARM builds as `Experimental`?

One piece of feedback is that we (and this may be recent) *do not guarantee* that you can upgrade from an Alpha/Beta/RC to the actual full release (as in, you might not be able to go from 22.1.X -> 22.2-alpha1 -> 22.2.0. I feel like warrants a callout *somewhere* in our docs, I'm just not sure where. At the very least, I think it should be called out here -- the question is whether it requires add'l callout

File: _includes/windows_warning.md
Comment (mwang1026) [inline _includes/windows_warning.md:2]: Will we have an ARM64 warning as well that it's Experimental?

**Review by mwang1026 — CHANGES_REQUESTED** [review]
A few pieces of feedback Thanks!

**Comment by stbof** [comment]
The intro bullets distinguish between Testing and Experimental, but I only see Testing in the heading.

**Comment by mdlinville** [comment]
> The intro bullets distinguish between Testing and Experimental, but I only see Testing in the heading and experimental is relegated to a note after you click a specific platform under the Testing heading. This doesn't seem very consistent to me. I would expect to see an Experimental heading.

That's because Testing releases are for a CRDB version, but Experimental releases are for all CRDB binaries for a given platform (like Windows or ARM). I'll add the Experimental labels to the Windows table to be consistent.

File: releases/index.md
Comment (mdlinville) [inline releases/index.md:160]:
```suggestion
            <td>Intel 64-bit (<b>Experimental</b>) Downloads</td>
```

File: releases/index.md
Comment (mdlinville) [inline releases/index.md:154]:
```suggestion
    Windows 8 or higher is required.

```
There's no reason for this to be in a warning.

File: releases/index.md
Comment (mdlinville) [reply releases/index.md:19]: Yes, it might be more clear if you look at the HTML preview.

**Review by stbof — APPROVED** [review]

File: releases/index.md
Comment (mwang1026) [reply releases/index.md:19]: Is there a way to refresh the preview with the latest changes? It's been a while since I reviewed a docs issue that needed a UX review as well!

When I clicked the preview from the PR description it did stick out to me that it's not super clear that Windows is Experimental (just a parenthetical on the column header of the link to download column).

Thoughts on either
(1) Have the windows election button have a denotation of experimental? Like that row with all the platforms to pick a download from
or
(2) Have the actual link itself say Experimental?

I'd assume it'd work the same way for Windows -- is there a way to generate a preview with ARM64? Might be hard given we're not publishing an ARM build yet

Comment (mwang1026) [reply releases/index.md:19]: I think part of what makes it weird is that Testing and Experimental are NOT mutually exclusive! Like, testing builds are also not qualified for production usage, in a sense. Unfortunately don't have concrete ideas to handle the cognitive dissonance here and OK if the answer is "you're thick, Michael"

Comment (mwang1026) [reply releases/index.md:19]: I think so. I wonder how many people read that though -- I guess we'll see? Not sure how else we denote it without making it too intrusive UX wise

Comment (mdlinville) [reply releases/index.md:19]: It's right at the top of the page. I think they would have to work hard to miss it? It's effectively exactly where the Windows one was before. I've added a popup alert to all experimental downloads. PTAL.

Comment (mdlinville) [reply releases/index.md:19]: I'm confused about why you're not seeing the preview. In the latest version, I added a Javascript alert for all Experimental download links (not the SHA256 links). This has been validating by the docs tooling folks, but does it help address your concern?

**Comment by mdlinville** [comment]
@jlinder @kenliu-crl The most recent commit update the Linux binary installation instructions as discussed. PTAL. I'd like this PR to go out tomorrow along with #15156 .

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:38]:
```suggestion
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:40]:
```suggestion
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:41]:
```suggestion
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:42]:
```suggestion
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:87]:
```suggestion
                <a href="{{ v.major_version }}.html#{{ r.version | replace: ".", "-" }}">{{ r.version }}</a> {% comment %} Add link to each release r. {% endcomment %}
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:133]:
```suggestion
                <a href="{{ v.major_version }}.html#{{ r.version | replace: ".", "-" }}">{{ r.version }}</a> {% comment %} Add link to each release r. {% endcomment %}
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:170]:
```suggestion
                <a href="{{ v.major_version }}.html#{{ r.version | replace: ".", "-" }}">{{ r.version }}</a> {% comment %} Add link to each release r. {% endcomment %}
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:210]:
```suggestion
                <a href="{{ v.major_version }}.html#{{ r.version | replace: ".", "-" }}">{{ r.version }}</a> {% comment %} Add link to each release r. {% endcomment %}
```

File: releases/index.md
Comment (nickvigilante) [inline releases/index.md:246]:
```suggestion
                <a href="{{ v.major_version }}.html#{{ r.version | replace: ".", "-" }}">{{ r.version }}</a> {% comment %} Add link to each release r. {% endcomment %}
```

**Review by nickvigilante — CHANGES_REQUESTED** [review]
Try that to start.

**Comment by mdlinville** [comment]
@mwang1026 There are a few pending items from you. PTAL. If I don't hear anything from you by tomorrow, we can address any open issues as a fast follow.

**Review by kenliu-crl — APPROVED** [review]
I reviewed this for content and structure, not for technical correctness. LGTM.

File: releases/index.md
Comment (mwang1026) [inline releases/index.md:284]: nit: s/`experimental download`/`experimental database deployment`

also small nit: Maybe add "Download will start when you close this dialogue" or something to the message as well? It wasn't clear to me as a user what to expect the first time when the dialogue popped up after I hit OK.

**Review by mwang1026 — APPROVED** [review]
:lgtm: Minor nit but don't want to hold up pushing the changes. Apologies for delay in review -- Roachfest kicked my butt in terms of time to do work

**Review by nickvigilante — APPROVED** [review]

File: v22.2/install-cockroachdb-linux.md
Comment (jlinder) [inline v22.2/install-cockroachdb-linux.md:45]:
```suggestion
         <li>Copy the library files to the directory. In the following commands, replace `{ARCHITECTURE}` with `linux-amd64` for Intel, or with `linux-3.7.10-gnu-aarch64` for ARM.</p>
```

For example, the file name for the alpha.4 archive is `cockroach-v22.2.0-alpha.4.linux-3.7.10-gnu-aarch64.tgz`.

File: _includes/releases/release-downloads-docker-image.md
Comment (jlinder) [inline _includes/releases/release-downloads-docker-image.md:41]: The filename format is similarly different for the cockroach-sql files. Here is the alpha.4 one: `cockroach-sql-v22.2.0-alpha.4.linux-3.7.10-gnu-aarch64.tgz`

**Review by jlinder — COMMENTED** [review]
A couple changes are needed because the archive names have a slightly different format. Once those are done, this looks good.

File: releases/index.md
Comment (mdlinville) [reply releases/index.md]: I ended up needing to add some info to the dialog and change it to a `confirm()` so that folks can decide **not** to download it.

**Comment by mdlinville** [comment]
@ianjevans had a look at the SCSS changes and approved them.

**Pull request merged**
